### PR TITLE
Initial Creation of document-generator-core module in document-generator

### PR DESCRIPTION
**Pull request by Mathsh**
updated pom.xml to include document-generator-core module
 - includes addition of spring-boot dependency.
Added document-generator-core module to project
 - updated new pom.xml to include junit, structured logging and kafka.

Added new package called kafka
Removed Consumer and Producer wrapper and replaced with a KafkaConfigHelper

Resolves SFA 421

### CODE REVIEW

File: document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/configurationHelper/KafkaConsumerProducerConfigHelper.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.companieshouse.document.generator.core.kafka.configurationHelper;
+
+
+import uk.gov.companieshouse.kafka.consumer.ConsumerConfig;
+import uk.gov.companieshouse.kafka.producer.ProducerConfig;
+
+public interface KafkaConsumerProducerConfigHelper {
```
Comment (jsimons1989) [inline document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/configurationHelper/KafkaConsumerProducerConfigHelper.java]: Would "KafkaBrokerConfigHelper" be more relevant to the behaviour of the interface (is it likely to contain more than these methods)? KafkaConfigHelper and KafkaConsumerProducerConfigHelper sound similar.

File: document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/configurationHelper/impl/KafkaConfigHelperImpl.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import uk.gov.companieshouse.kafka.consumer.ConsumerConfig;
+import uk.gov.companieshouse.kafka.producer.Acks;
+import uk.gov.companieshouse.kafka.producer.ProducerConfig;
+
+import java.util.List;
+
+public class KafkaConfigHelperImpl implements KafkaConfigHelper {
+
+    @Autowired
+    private KafkaConsumerProducerConfigHelper consumerProducerConfigHelperHelper;
```
Comment (jsimons1989) [inline document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/configurationHelper/impl/KafkaConfigHelperImpl.java]: Typo on "consumerProducerConfigHelperHelper". You could shorten to "brokerConfigHelper".

File: pom.xml [hunk 3]
```diff
@@ -32,7 +39,11 @@
     </properties>
 
     <dependencies>
-      <!-- As document-generator is progressed depedencies will be added -->
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-hateoas</artifactId>
```
Comment (jsimons1989) [inline pom.xml]: What is this being used for?

Comment (Mathsh) [reply pom.xml]: addition of spring boot to the parent pom. required me to add the dependancy.

File: document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/configurationHelper/KafkaBrokerConfigHelper.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.companieshouse.document.generator.core.kafka.configurationHelper;
```
Comment (ch-mcabrera) [inline document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/kafka/configurationHelper/KafkaBrokerConfigHelper.java]: Package name should be in lower case, shouldnt it?

**Pull request merged**